### ISSUE TITLE
Eclipse 2024-03: An internal error occurred during: "Initialize CompilationUnit Manager"

### ISSUE THREAD

**Issue by kriegaex** [eclipse-aspectj/ajdt#49]
```text
An internal error occurred during: "Initialize CompilationUnit Manager".
class org.eclipse.ajdt.core.javaelements.AspectJMemberElement overrides final method org.eclipse.jdt.internal.core.JavaElement.hashCode()I
```

This is caused by yet another round of breaking changes in JDT Core. The fix probably involves

  * refreshing AspectJ JDT Core from upstream,
  * making `JavaElement.hashCode()` non-final,
  * fixing other similar problems,
  * building and releasing a new JDT Core version,
  * adjusting AspectJ to the new JDT Core,
  * adjusting AJDT to the new JDT Core packaged into AspectJ,
  * creating a new AJDT update site for 2024-03,
  * updating the Eclipse Marketplace entry for AJDT to reflect that is is fit for 2024-03.

Probably, this sysiphus-like effort makes most sense when upgrading JDT Core to Java 22 anyway, whenever Java 22 support will be merged into upstream JDT Core in a halfway stable and feature-complete version. In the last few release cycles, new Java version feature support was clearly incomplete after the merge.

**Comment by daniel-matheis-vivavis** [comment]
"this sysiphus-like effort makes most sense when upgrading JDT Core to Java 22 anyway"
When do you think you will have time for this?
We wanted to upgrade to JDK 21 and Eclipse 2024-03 as target platform for our RCP application and are blocked by this.

**Comment by kriegaex** [comment]
It is done when it is done. You can use JDK 21 on Eclipse 2023-12 for now.

The thing is, in order to fix this for AJDT, I also need to merge in all the upstream JDT Core stuff into AspectJ, necessitating yet another AspectJ release, which might break other stuff. I want to do that only once for JDK 22, which then of course will also work for JDK 21. If you are so bleeding-edge that you think you need to upgrade to 2024-03 at the first chance, you simply need to wait for now, if you think 2023-12 is not an option for you.

**Comment by kriegaex** [comment]
Please try https://download.eclipse.org/tools/aspectj/ajdt/431/dev/update/ or install from the Eclipse Marketplace, to where I just added the new update site.

**Comment by akuhtz** [comment]
@kriegaex Thank you for the update of AJDT :+1:

**Comment by daniel-matheis-vivavis** [comment]
@kriegaex  Same from here: Thanks a lot for your work!

And I just can repeat what i wrote:
> Of course, I understand that it is your free time and appreciate your effort no matter if you find the time now or not...

So please in the future don't feel urged or forced to answer, to fix something or whatever you have felt...
At least to me, you don't have to justify yourself!
Have a nice day.
